src/base/model.js

```diff
@@ -243,16 +243,29 @@ ModelBase.prototype.isNew = function() {
  * @returns {Object} Serialized model as a plain object.
  */
 ModelBase.prototype.serialize = function(options = {}) {
-  const { shallow = false, omitPivot = false, ...rest } = options;
+  const { shallow = false, omitPivot = false, omitNew = false, ...rest } = options;
   const { attributes } = this;
 
+  if (omitNew && this.isNew()) {
+    return null;
+  }
+
   if (!shallow) {
 
-    const relations = mapValues(this.relations, (relation, key) =>
-      relation.toJSON != null
-        ? relation.toJSON({ shallow, omitPivot, ...rest })
-        : relation
-    );
+    let relations = mapValues(this.relations, (relation, key) => {
+      if (relation.toJSON != null) {
+        let rel = relation.toJSON(options);
+        if (_.isArray(rel)) {
+          // Filter null values in a collection coming from the omitNew option
+          rel = _.filter(rel, x => ! _.isNull(x));
+        }
+        return rel;
+      }
+        relation
```

Comment:
As well array filtering shall be done in [`base/collection.js`](https://github.com/tgriesser/bookshelf/blob/master/src/base/collection.js#L120-L155) instead of here.

[CollectionBase#serialize](https://github.com/tgriesser/bookshelf/blob/master/src/base/collection.js#L120-L138) docs shall be updated as well.